pandas/core/groupby.py

```diff
@@ -1807,6 +1811,11 @@ def wrapper(*args, **kwargs):
     def _cython_operation(self, kind, values, how, axis):
         assert kind in ['transform', 'aggregate']
 
+        # categoricals are only 1d, so we
+        # are not setup for dim transforming
+        if is_categorical_dtype(values):
+            raise NotImplementedError
```

Comment:
According to the coverage report, this error is never reached. Is it needed for the categorical you added in the test?